### PR TITLE
More gun variants, part the third.

### PR DESCRIPTION
**Pull request by DoctorBoomstick**
#### Summary
None

#### Purpose of change
Continues work started in #61598 and #61893, carrying on the inexorable march towards extending the variants system across more firearms which currently lack a more generic variation.

#### Describe the solution
I’ve created variants for all the weapons in .22 calibre, 7.62x25mm, 9x18mm, and 5.7x28mm. I’ve also slightly adjusted the existing variants present across the 5.7mm guns (the FN P90, Kel-Teck P50, and FN PS90), to make the names more compact and the descriptions less wordy. In doing these changes, many of the magazines have also been given appropriate variants were applicable.

#### Describe alternatives you've considered
No real alternatives to speak of, save for perhaps finding ways to cut down on the number of magazine variants.

#### Testing
All elements have been bug tested on the latest experimental builds, with the game loading just fine.
#### Additional context
Consult #61598 and #61893 for the first two parts to this expansion of the existing gun variants.

### CODE REVIEW

**Review by MNG-cataclysm — APPROVED** [review]

**Review by anothersimulacrum — APPROVED** [review]

File: data/json/items/gun/22.json [hunk 1]
```diff
@@ -178,8 +205,17 @@
     "looks_like": "ar15",
     "type": "GUN",
     "reload_noise_volume": 10,
-    "name": { "str_sp": "Ruger 10/22" },
-    "description": "A combination of negligible recoil, high customization, and inexpensive ammunition made the Ruger 10/22 a popular choice among enthusiasts and small game hunters.  Even though it fires the weak .22 round, it's perfectly capable of killing.",
+    "name": { "str": "varmint rifle" },
+    "description": "a lightweight and modular pistol calibre hunting carbine, firing the low power .22 LR cartridge.  Before the Cataclysm, a mix of high customizability, cheep ammunition, and low recoil made this a very popular varmint rifle amongst civilian shooters.",
```
Comment (TheShadowFerret) [inline data/json/items/gun/22.json:209]:
```suggestion
    "description": "A lightweight and modular pistol caliber hunting carbine, firing the low power .22 LR cartridge.  Before the Cataclysm, a mix of high customizability, cheep ammunition, and low recoil made this a very popular varmint rifle amongst civilian shooters.",
```

File: data/json/items/gun/22.json [hunk 2]
```diff
@@ -242,8 +288,20 @@
     "looks_like": "glock_17",
     "type": "GUN",
     "reload_noise_volume": 10,
-    "name": { "str_sp": "SIG Mosquito" },
-    "description": "A popular, small .22 LR pistol.  One of the many polymer frame pistols produced during the late 20th and early 21st centuries; it has an underbarrel rail.",
+    "name": { "str": "Mosquito pistol" },
+    "description": "a compact, .22 calibre semi-automatic pistol.  Built from aluminum and polymer componence, the weapon is light weight and concealable.  Unfortunately, much like its namesake, this pistol mostly poses a mild annoyance to most hostiles that the Cataclysm is liable to direct into your path.",
```
Comment (TheShadowFerret) [inline data/json/items/gun/22.json:292]:
```suggestion
    "description": "a compact, .22 caliber semi-automatic pistol.  Built from aluminum and polymer componence, the weapon is lightweight and concealable.  Unfortunately, much like its namesake, this pistol mostly poses a mild annoyance to most hostiles that the Cataclysm is liable to direct into your path.",
```

File: data/json/items/gun/22.json [hunk 3]
```diff
@@ -304,11 +371,20 @@
   },
   {
     "id": "j22",
-    "copy-from": "pistol_backup",
     "looks_like": "glock_17",
     "type": "GUN",
-    "name": { "str": "Jennings J-22" },
-    "description": "One of the quintessential 'Saturday night specials', the Jennings J-22 was very affordably priced with its injection-molded zinc slide and frame.  Intended to fill the void left after small pocket pistols were banned from import, these were more commonly used by criminals unfazed by their glaring safety issues.",
+    "name": { "str": "cheap .22 calibre pistol" },
+    "description": "Made from only the cheapest of budget metals, firing one of the weakest cartridges, and having a hearty laugh at the concept of user safety, this .22 calibre pocket pistol is of questionable utility at best.  Affordably priced before the Cataclysm, the pistol’s price point was arguably its only positive element, an element which pertains zero bearing on your current situation.",
```
Comment (TheShadowFerret) [inline data/json/items/gun/22.json:377]:
```suggestion
    "name": { "str": "cheap .22 caliber pistol" },
    "description": "Made from only the cheapest of budget metals, firing one of the weakest cartridges, and having a hearty laugh at the concept of user safety, this .22 caliber pocket pistol is of questionable utility at best.  Affordably priced before the Cataclysm, the pistol’s price point was arguably its only positive element, an element which pertains zero bearing on your current situation.",
```

File: data/json/items/gun/57.json [hunk 4]
```diff
@@ -4,8 +4,17 @@
     "looks_like": "glock_17",
     "type": "GUN",
     "reload_noise_volume": 10,
-    "name": { "str": "FN Five-seveN" },
-    "description": "Designed to work with FN's proprietary 5.7x28mm round, the Five-seveN is a lightweight pistol with a very high capacity, best used against armored opponents.",
+    "name": { "str": "5.7mm pistol" },
+    "description": "Firing the armour-piercing 5.7x28mm round, this light-weight handgun was developed for NATO use as a sidearm capable of penetrating body armour.  Its proprietary and nonprolific ammunition kept it comparatively rare amongst both civilian and militant use, but the gun’s ability to ignore most forms of conventional ballistic armour can’t be argued with.  Given that its tiny bullets have next to no stopping power, its armour-penetrating capability comes at a considerable trade-off.",
```
Comment (TheShadowFerret) [inline data/json/items/gun/57.json:8]:
```suggestion
    "description": "Firing the armor-piercing 5.7x28mm round, this lightweight handgun was developed for NATO use as a sidearm capable of penetrating body armor.  Its proprietary and nonprolific ammunition kept it comparatively rare amongst both civilian and militant use, but the gun’s ability to ignore most forms of conventional ballistic armor can’t be argued with.  Given that its tiny bullets have next to no stopping power, its armor-penetrating capability comes at a considerable trade-off.",
```

File: data/json/items/gun/57.json [hunk 5]
```diff
@@ -44,14 +53,14 @@
     "looks_like": "hk_mp5",
     "type": "GUN",
     "reload_noise_volume": 10,
-    "name": { "str": "FN 5.7x28mm PDW" },
-    "description": "An FN manufactured personal defence weapon, this automatic firearm is chambered for the high-velocity proprietary 5.7x28mm cartridge.  Accepting top-mounted 50-round custom magazines which are fitted to the receiver of the weapon behind the grip, thus making it a bullpup design, this little submachine gun packs quite the nasty payload.  Its comparatively small, high-speed caliber makes it excellent at both defeating armoured opponents and firing short bursts manageably.",
+    "name": { "str": "5.7mm automatic PDW" },
+    "description": "A late 90s manufactured personal defence weapon (PDW), this automatic firearm fires the high-velocity and proprietary 5.7x28mm cartridge.  Making use of top-mounted 50-round custom magazines which lock into the receiver of the weapon behind the grip, this little submachine gun packs quite the nasty payload.  Its small, high-speed calibre makes it excellent at both defeating armoured opponents and firing short bursts manageably.",
```
Comment (TheShadowFerret) [inline data/json/items/gun/57.json:57]:
```suggestion
    "description": "A late 90s manufactured personal defense weapon (PDW), this automatic firearm fires the high-velocity, proprietary 5.7x28mm cartridge.  Making use of top-mounted 50-round custom magazines which lock into the receiver of the weapon behind the grip, this little submachine gun packs quite the nasty payload.  Its small, high-speed caliber makes it excellent at both defeating armored opponents and firing short bursts manageably.",
```

File: data/json/items/gun/57.json [hunk 6]
```diff
@@ -44,14 +53,14 @@
...
+        "description": "The first in a new genre of guns, termed \"personal defence weapons.\"  FN designed the P90 to use their proprietary high-velocity 5.7x28mm ammunition.  Made to fulfill NATO requests for a weapon chambered in a more suitable cartridge for defeating body armour than standard 9x19mm submachine guns, the FN P90 is excellent at firing short and controllable bursts manageably.  The PDW’s small and high-speed 5.7mm ammunition, of which 50 rounds fit in the P90’s top mounted magazine, packs quite the lethal payload for any armoured opponent, whilst the firearm’s bullpup design makes it commendably compact to boot.",
```
Comment (TheShadowFerret) [inline data/json/items/gun/57.json:63]:
```suggestion
        "description": "The first in a new genre of guns, termed \"personal defense weapons.\"  FN designed the P90 to use their proprietary high-velocity 5.7x28mm ammunition.  Made to fulfill NATO requests for a weapon chambered in a more suitable cartridge for defeating body armor than standard 9x19mm submachine guns, the FN P90 is excellent at firing short and controllable bursts manageably.  The PDW’s small and high-speed 5.7mm ammunition, of which 50 rounds fit in the P90’s top mounted magazine, packs quite the lethal payload for any armored opponent, whilst the firearm’s bullpup design makes it commendably compact to boot.",
```

File: data/json/items/gun/57.json [hunk 7]
```diff
@@ -137,8 +146,17 @@
     "looks_like": "glock_17",
     "type": "GUN",
     "reload_noise_volume": 10,
-    "name": { "str": "Ruger-57" },
-    "description": "A full-sized polymer framed handgun chambered in the unusual 5.7x28mm cartridge, the Ruger-57 was one of the first commercially available 5.7mm pistols on the market, with its main competitor being FN’s decades-old, and far pricier, Five-seveN pistol.  Featuring a lightweight ergonomic design, 1911-style thumb-safety, a high visibility green front-sight with adjustable black rear iron-sights, and a price point which hovered a good 500 dollars below its competition, the Ruger-57 made the comparatively niche 5.7x28mm round far more accessible to a wider margin of potential shooters.",
+    "name": { "str": "Ruger-57 handgun" },
+    "description": "Produced by the Ruger gun company, this full-sized pistol makes use of the 5.7x28mm cartridge.  one of the first pistols to both make use of the unusual round and achieve market success within the civilian world, it opened the door for accessible 5.7mm shooting due to its low price and reliable design.",
```
Comment (TheShadowFerret) [inline data/json/items/gun/57.json:150]:
```suggestion
    "description": "Produced by the Ruger gun company, this full-sized pistol makes use of the 5.7x28mm cartridge.  One of the first pistols to both make use of the unusual round and achieve market success within the civilian world, it opened the door for accessible 5.7mm shooting due to its low price and reliable design.",
```

File: data/json/items/gun/762x25.json [hunk 8]
```diff
@@ -4,8 +4,17 @@
     "looks_like": "hk_mp5",
     "type": "GUN",
     "reload_noise_volume": 10,
-    "name": { "str": "PPSh-41" },
-    "description": "The Soviet-made PPSh-41 is a mass-produced selective-fire submachine gun.  It has a relatively high rate of fire.",
+    "name": { "str": "PPSh-41 submachine gun" },
+    "description": "A bulky and simple Second World War submachine gun of Russian manufacture, composed from a basic wooden buttstock, steel barrel shroud, and the hopes and prayers of soviet soldiers under threat of German invasion.  Designed for mass production with limited materials, the weapon is notable for its high-capacity drum magazines, capability of rapidly depleting said magazines, and vicious 7.62x25mm chambering.",
```
Comment (TheShadowFerret) [inline data/json/items/gun/762x25.json:8]:
```suggestion
    "description": "A bulky and simple Second World War submachine gun of Russian manufacture, composed from a basic wooden buttstock, steel barrel shroud, and the hopes and prayers of Soviet soldiers under threat of German invasion.  Designed for mass production with limited materials, the weapon is notable for its high-capacity drum magazines, capability of rapidly depleting said magazines, and vicious 7.62x25mm chambering.",
```

File: data/json/items/gun/762x25.json [hunk 9]
```diff
@@ -4,8 +4,17 @@
...
+    "variant_type": "gun",
+    "variants": [
+      {
+        "id": "ppsh",
+        "name": { "str": "PPSh-41" },
+        "description": "The Soviet-made PPSh-41 is a mass-produced selective-fire submachine gun.  It has a relatively high rate of fire.",
```
Comment (TheShadowFerret) [inline data/json/items/gun/762x25.json:14]: Why is the non-generic description less detailed than the generic one 🤔

File: data/json/items/gun/9x18.json [hunk 10]
```diff
@@ -4,9 +4,18 @@
     "copy-from": "pistol_base",
     "looks_like": "glock_17",
     "type": "GUN",
-    "name": { "str": "Makarov PM" },
-    "//": "10 and 12-round magazines exist, but were made for the newer PMM, and are not compatible with the old PM.",
-    "description": "The Pistolet Makarova was developed by the Soviet Union to replace the WWII-era TT-33 pistol.  It uses the 9x18mm cartridge, which remains in use among various former Soviet countries.",
+    "name": { "str": "Makarov pistol" },
+    "description": "A svelte, Cold War era handgun of Russian heritage, this former military pistol fires the Soviet 9x18mm cartridge.  Constructed completely out of steel components and experiencing relative popularity throughout eastern European countries, it was a favoured backup weapon amongst USSR security forces, soldiers, and covert units.",
```
Comment (TheShadowFerret) [inline data/json/items/gun/9x18.json:8]:
```suggestion
    "description": "A svelte, Cold War era handgun of Russian heritage, this former military pistol fires the Soviet 9x18mm cartridge.  Constructed completely out of steel components and experiencing relative popularity throughout eastern European countries, it was a favored backup weapon amongst USSR security forces, soldiers, and covert units.",
```

File: data/json/items/gun/9x18.json [hunk 11]
```diff
@@ -4,9 +4,18 @@
...
+    "variant_type": "gun",
+    "variants": [
+      {
+        "id": "makarov",
+        "name": { "str": "Makarov PM" },
+        "//": "10 and 12-round magazines exist, but were made for the newer PMM, and are not compatible with the old PM.",
+        "description": "The Pistolet Makarova was developed by the Soviet Union to replace the WWII-era TT-33 pistol.  It uses the 9x18mm cartridge, which remains in use among various former Soviet countries.",
```
Comment (TheShadowFerret) [inline data/json/items/gun/9x18.json:15]: This description is also more brief than the generic version 🤔

File: data/json/items/gun/9x18.json [hunk 12]
```diff
@@ -40,9 +50,18 @@
     "copy-from": "skorpion_61",
     "looks_like": "hk_mp5",
     "type": "GUN",
-    "name": { "str": "Škorpion vz. 82" },
-    "//": "Total unloaded weight of gun 1451.496 grams, rounded to 1451.  Current weight of folding stock 200 grams.",
-    "description": "A version of the Škorpion submachine gun chambered in 9x18mm Makarov, with a slightly longer barrel than the original design.",
+    "name": { "str": "vz. 82 submachine gun" },
+    "description": "A compact submachine gun hailing from Czechoslovakia during the Cold War, featuring a suppressively high rate of automatic fire and shooting the soviet 9x18mm round.",
```
Comment (TheShadowFerret) [inline data/json/items/gun/9x18.json:54]:
```suggestion
    "description": "A compact submachine gun hailing from Czechoslovakia during the Cold War, featuring a suppressively high rate of automatic fire and shooting the Soviet 9x18mm round.",
```

File: data/json/items/magazine/22.json [hunk 13]
```diff
@@ -20,8 +20,17 @@
     "id": "a180mag",
     "looks_like": "mp5mag",
     "type": "MAGAZINE",
-    "name": { "str": "American-180 165-round magazine" },
-    "description": "An unusually shaped pan magazine for the American-180.",
+    "name": { "str": ".22 165-round magazine" },
+    "description": "an unusually shaped pan magazine for use with .22 calibre submachine guns.",
```
Comment (TheShadowFerret) [inline data/json/items/magazine/22.json:24]:
```suggestion
    "description": "An unusually-shaped pan magazine for use with .22 caliber submachine guns.",
```

File: data/json/items/magazine/22.json [hunk 14]
```diff
@@ -20,8 +20,17 @@
...
+    "variant_type": "gun",
+    "variants": [
+      {
+        "id": "a180mag",
+        "name": { "str": "American-180 165-round magazine" },
+        "description": "An unusually shaped pan magazine for the American-180.",
```
Comment (TheShadowFerret) [inline data/json/items/magazine/22.json:30]:
```suggestion
        "description": "An unusually-shaped pan magazine for the American-180.",
```

File: data/json/items/magazine/22.json [hunk 15]
```diff
@@ -38,8 +47,17 @@
     "id": "a180mag1",
     "looks_like": "mp5mag",
     "type": "MAGAZINE",
-    "name": { "str": "American-180 polymer 165-round magazine" },
-    "description": "An unusually shaped pan magazine for the American-180, made of polymer.",
+    "name": { "str": ".22 165-round polymer magazine" },
+    "description": "an unusually shaped polymer constructed pan magazine for use with .22 calibre submachine guns.",
```
Comment (TheShadowFerret) [inline data/json/items/magazine/22.json:51]:
```suggestion
    "description": "An unusually-shaped polymer constructed pan magazine for use with .22 caliber submachine guns.",
```

File: data/json/items/magazine/22.json [hunk 16]
```diff
@@ -38,8 +47,17 @@
...
+    "variant_type": "gun",
+    "variants": [
+      {
+        "id": "a180mag1",
+        "name": { "str": "American-180 polymer 165-round magazine" },
+        "description": "An unusually shaped pan magazine for the American-180, made of polymer.",
```
Comment (TheShadowFerret) [inline data/json/items/magazine/22.json:57]:
```suggestion
        "description": "An unusually-shaped, polymer pan magazine for the American-180.",
```

File: data/json/items/magazine/22.json [hunk 17]
```diff
@@ -56,8 +74,17 @@
     "id": "a180mag2",
     "looks_like": "mp5mag",
     "type": "MAGAZINE",
-    "name": { "str": "American-180 177-round magazine" },
-    "description": "An unusually shaped pan magazine for the American-180",
+    "name": { "str": ".22 177-round magazine" },
+    "description": "an unusually shaped pan magazine for use with .22 calibre submachine guns.",
```
Comment (TheShadowFerret) [inline data/json/items/magazine/22.json:78]:
```suggestion
    "description": "An unusually-shaped pan magazine for use with .22 caliber submachine guns.",
```

File: data/json/items/magazine/22.json [hunk 18]
```diff
@@ -56,8 +74,17 @@
...
+    "variant_type": "gun",
+    "variants": [
+      {
+        "id": "a180mag2",
+        "name": { "str": "American-180 177-round magazine" },
+        "description": "An unusually shaped pan magazine for the American-180",
```
Comment (TheShadowFerret) [inline data/json/items/magazine/22.json:84]:
```suggestion
        "description": "An unusually-shaped pan magazine for the American-180.",
```

File: data/json/items/magazine/22.json [hunk 19]
```diff
@@ -74,8 +101,17 @@
     "id": "a180mag3",
     "looks_like": "mp5mag",
     "type": "MAGAZINE",
-    "name": { "str": "American-180 polymer 220-round magazine" },
-    "description": "An unusually shaped pan magazine for the American-180, made of polymer.",
+    "name": { "str": ".22 220-round polymer magazine" },
+    "description": "an unusually shaped polymer constructed pan magazine for use with .22 calibre submachine guns.",
```
Comment (TheShadowFerret) [inline data/json/items/magazine/22.json:105]:
```suggestion
    "description": "An unusually-shaped polymer pan magazine for use with .22 caliber submachine guns.",
```

File: data/json/items/magazine/22.json [hunk 20]
```diff
@@ -74,8 +101,17 @@
...
+    "variant_type": "gun",
+    "variants": [
+      {
+        "id": "a180mag3",
+        "name": { "str": "American-180 polymer 220-round magazine" },
+        "description": "An unusually shaped pan magazine for the American-180, made of polymer.",
```
Comment (TheShadowFerret) [inline data/json/items/magazine/22.json:111]:
```suggestion
        "description": "An unusually-shaped polymer pan magazine for the American-180.",
```

File: data/json/items/magazine/22.json [hunk 21]
```diff
@@ -92,8 +128,17 @@
...
-    "name": { "str": "American-180 polymer 275-round magazine" },
-    "description": "An unusually shaped pan magazine for the American-180, made of polymer.",
+    "name": { "str": ".22 275-round polymer magazine" },
+    "description": "an unusually shaped polymer constructed pan magazine for use with .22 calibre submachine guns.",
+    "variant_type": "gun",
+    "variants": [
+      {
+        "id": "a180mag4",
+        "name": { "str": "American-180 polymer 275-round magazine" },
+        "description": "An unusually shaped pan magazine for the American-180, made of polymer.",
```
Comment (TheShadowFerret) [inline data/json/items/magazine/22.json:138]:
```suggestion
        "description": "An unusually-shaped polymer pan magazine for the American-180.",
```

File: data/json/items/magazine/22.json [hunk 22]
```diff
@@ -92,8 +128,17 @@
     "id": "a180mag4",
     "looks_like": "mp5mag",
     "type": "MAGAZINE",
-    "name": { "str": "American-180 polymer 275-round magazine" },
-    "description": "An unusually shaped pan magazine for the American-180, made of polymer.",
+    "name": { "str": ".22 275-round polymer magazine" },
+    "description": "an unusually shaped polymer constructed pan magazine for use with .22 calibre submachine guns.",
```
Comment (TheShadowFerret) [inline data/json/items/magazine/22.json:132]:
```suggestion
    "description": "An unusually-shaped polymer pan magazine for use with .22 caliber submachine guns.",
```

File: data/json/items/magazine/22.json [hunk 23]
```diff
@@ -144,8 +198,17 @@
     "id": "ruger1022bigmag",
     "looks_like": "stanag30",
     "type": "MAGAZINE",
-    "name": { "str": "Ruger 10/22 25-round magazine" },
-    "description": "An extended 25-round stick magazine for the Ruger 10/22, known as the BX-25.",
+    "name": { "str": "25-round varmint magazine" },
+    "description": "An extended capacity 25-round magazine, for use with .22 calibre varmint rifles.",
```
Comment (TheShadowFerret) [inline data/json/items/magazine/22.json:202]:
```suggestion
    "description": "An extended capacity 25-round magazine, for use with .22 caliber varmint rifles.",
```

File: data/json/items/magazine/22.json [hunk 24]
```diff
@@ -161,8 +224,17 @@
     "id": "ruger1022mag",
     "looks_like": "stanag30",
     "type": "MAGAZINE",
-    "name": { "str": "Ruger 10/22 10-round rotary magazine" },
-    "description": "A removable 10-round rotary magazine for the Ruger 10/22 which allows the magazine to fit flush with the bottom of the stock.",
+    "name": { "str": "10-round varmint magazine" },
+    "description": "A flush fitting 10-round rotary magazine, for use with .22 calibre varmint rifles.",
```
Comment (TheShadowFerret) [inline data/json/items/magazine/22.json:228]:
```suggestion
    "description": "A flush fitting 10-round rotary magazine, for use with .22 caliber varmint rifles.",
```

File: data/json/items/magazine/22.json [hunk 25]
```diff
@@ -179,8 +251,17 @@
     "id": "sw22mag",
     "looks_like": "glock17_17",
     "type": "MAGAZINE",
-    "name": { "str": "22A 10-round magazine" },
-    "description": "A standard capacity magazine for the popular S&W 22A pistol.",
+    "name": { "str": ".22 10-round S&W magazine" },
+    "description": "A 10-round handgun magazine for use with Smith and Wesson’s .22 calibre pocket pistol.",
```
Comment (TheShadowFerret) [inline data/json/items/magazine/22.json:255]:
```suggestion
    "description": "A 10-round handgun magazine for use with Smith and Wesson’s .22 caliber pocket pistol.",
```

File: data/json/items/magazine/22.json [hunk 26]
```diff
@@ -196,8 +277,17 @@
     "id": "j22mag",
     "looks_like": "glock17_17",
     "type": "MAGAZINE",
-    "name": { "str": "J-22 6-round magazine" },
-    "description": "A cheap 6-round steel box magazine for the Jennings J-22.",
+    "name": { "str": "cheap 6-round magazine" },
+    "description": "A cheaply built, low quality 6-round magazine, for use in only the most ineffectual of .22 calibre pocket pistols.",
```
Comment (TheShadowFerret) [inline data/json/items/magazine/22.json:281]:
```suggestion
    "description": "A cheaply built, low quality 6-round magazine, for use in only the most ineffectual of .22 caliber pocket pistols.",
```

File: data/json/items/magazine/22.json [hunk 27]
```diff
@@ -213,8 +303,17 @@
     "id": "wp22mag",
     "looks_like": "glock17_17",
     "type": "MAGAZINE",
-    "name": { "str": "P22 10-round magazine" },
-    "description": "A 10-round box magazine for the Walther P22.",
+    "name": { "str": ".22 10-round Walther magazine" },
+    "description": "A 10-round handgun magazine for use with Walther's .22 calibre pocket pistol.",
```
Comment (TheShadowFerret) [inline data/json/items/magazine/22.json:307]:
```suggestion
    "description": "A 10-round handgun magazine for use with Walther's .22 caliber pocket pistol.",
```

File: data/json/items/magazine/57.json [hunk 28]
```diff
@@ -3,8 +3,17 @@
...
-    "name": { "str": "Five-seveN 20-round magazine" },
-    "description": "A standard capacity magazine for use with the FN Five-seveN.",
+    "name": { "str": "5.7mm 20-round pistol magazine" },
+    "description": "A 20-round box magazine, for use with handguns chambering the proprietary 5.7x28mm cartridge.",
+    "variant_type": "gun",
+    "variants": [
+      {
+        "id": "fn57mag",
+        "name": { "str": "Five-seveN 20-round magazine" },
+        "description": "A standard capacity magazine for use with the FN Five-seveN.",
```
Comment (TheShadowFerret) [inline data/json/items/magazine/57.json:13]:
```suggestion
        "description": "A standard-capacity magazine for use with the FN Five-seveN.",
```
Many New England states would consider this legally highcap(FN makes a ten rounder for those states). It is standard for the firearm tho

**Review by TheShadowFerret — CHANGES_REQUESTED** [review]
~~Who dares bring British spellings to my American apocalypse~~

**Comment by nopenoperson** [comment]
i read the first pr and i dont really understand the problem that this is trying to fix.  also the variant system makes sense for t shirts, not for mosin nagants made 100 years apart with wildly different materials and qa standards.  functionally different guns should have functionally different stats!

File: data/json/items/gun/22.json [hunk 29]
```diff
@@ -4,8 +4,17 @@
     "looks_like": "hk_mp5",
     "type": "GUN",
     "reload_noise_volume": 10,
-    "name": { "str_sp": "American-180" },
-    "description": "The American-180 is a submachine gun developed in the 1960s that fires .22 LR, unusual for an SMG.  Though the round is low-powered, the high rate of fire and large magazine makes the 180 a formidable weapon.",
+    "name": { "str": ".22 calibre submachine gun" },
```
Comment (TheShadowFerret) [inline data/json/items/gun/22.json:7]:
```suggestion
    "name": { "str": ".22 caliber submachine gun" },
```

File: data/json/items/gun/22.json [hunk 30]
```diff
@@ -49,8 +58,17 @@
     "looks_like": "ar15",
     "type": "GUN",
     "reload_noise_volume": 10,
-    "name": { "str_sp": "Marlin 39A" },
-    "description": "The oldest and longest-produced shoulder firearm in the world.  Though it fires the weak .22 round, it is highly accurate and damaging, and has essentially no recoil.",
+    "name": { "str": "small game rifle" },
+    "description": "A venerable .22 calibre lever-action rifle, with a 19-round tube magazine and a classic design that dates back over a century.  Generally used for the hunting of small game like rabbits, the strict taxonomic classification of zombies as 'small game' is debatable, but this decades old weapon can still be counted upon to help manage the undead population.",
```
Comment (TheShadowFerret) [inline data/json/items/gun/22.json:62]:
```suggestion
    "description": "A venerable .22 caliber lever-action rifle, with a 19-round tube magazine and a classic design that dates back over a century.  Generally used for the hunting of small game like rabbits, the strict taxonomic classification of zombies as 'small game' is debatable, but this decades-old weapon can still be counted upon to help manage the undead population.",
```

File: data/json/items/gun/22.json [hunk 31]
```diff
@@ -242,8 +288,20 @@
     "looks_like": "glock_17",
     "type": "GUN",
     "reload_noise_volume": 10,
-    "name": { "str_sp": "SIG Mosquito" },
-    "description": "A popular, small .22 LR pistol.  One of the many polymer frame pistols produced during the late 20th and early 21st centuries; it has an underbarrel rail.",
+    "name": { "str": "Mosquito pistol" },
+    "description": "a compact, .22 caliber semi-automatic pistol.  Built from aluminum and polymer componence, the weapon is lightweight and concealable.  Unfortunately, much like its namesake, this pistol mostly poses a mild annoyance to most hostiles that the Cataclysm is liable to direct into your path.",
```
Comment (TheShadowFerret) [inline data/json/items/gun/22.json:292]:
```suggestion
    "description": "A compact, .22 caliber semi-automatic pistol.  Built from aluminum and polymer componence, the weapon is lightweight and concealable.  Unfortunately, much like its namesake, this pistol mostly poses a mild annoyance to most hostiles that the Cataclysm is liable to direct into your path.",
```

**Review by TheShadowFerret — CHANGES_REQUESTED** [review]

**Comment by DoctorBoomstick** [comment]
@nopenoperson, I’m not the one who implemented gun variants from the onset, so I might get the reasoning wrong and/ or fuck up the intended point. As such, I’d advise consulting Pr #46814 for further information about the system. That being said, as far as I understand, gun variants are aimed at cutting down on the gigantic number of similar, near-identical, functionally the same guns we currently have in the game. For some folks, coming across five different types of assault rifles that are basically the same save in name and description can be overwhelming, especially if they’re not versed in some basic gun terminology. The system can be toggled on to simplify matters for non-gun people who do not want to deal with many different types of practically the same gun, and/ or who don’t understand the weapon gun nut talk present in some of the descriptions and name. on the other hand, if you enjoy rolling around in heaps of different guns Scrooge Mick Duck style, you can just leave it off and have your armoury’s Werth of different guns.
Why do variants make sense for t-shirts? Simply cause all it’s doing is changing around descriptors, an inconsequential element. Why does gun variants make sense for guns? Simply because it sometimes takes near-identical weapons gameplay wise, like the M9 and 90two, were the only real differences are a name, description, and a few teeny tiny grams of weight and ML of volume, and puts them together under one name. again, if you don’t like this, you can just turn it off. For types of variants like the ones added in this PR, you do not even have any merging going on. It literally just makes the names and descriptions simpler, easier to understand, and holds the player’s hand a little if they don’t know what some gun terminologies are meant to mean.

**Comment by bombasticSlacks** [comment]
This looks good to me, could you add the barrel lengths to the guns? #64961 gives an example.

**Pull request merged**